crypto/evp/ctrl_params_translate.c

```diff
@@ -631,7 +631,7 @@ static int default_fixup_args(enum state state,
                     return 1;
                 case OSSL_PARAM_UTF8_STRING:
                     return OSSL_PARAM_get_utf8_string(ctx->params,
-                                                      ctx->p2, ctx->sz);
+                                                      (char **)&ctx->p2, ctx->sz);
                 case OSSL_PARAM_OCTET_STRING:
                     return OSSL_PARAM_get_octet_string(ctx->params,
```

Comment:
So I wonder, why doesn't `OSSL_PARAM_OCTET_STRING` get the same treatment?

Reply:
IMO, and I could totally be wrong, but it seems to me that the reason why `OSSL_PARAM_OCTET_STRING` does not get the same treatment is because all the translations related to it in [`evp_pkey_ctx_translations`](https://github.com/openssl/openssl/blob/1adc45b1ded7072d1566addeab227efa81b6c947/crypto/evp/ctrl_params_translate.c#L2045) do no have a `fix_up_args` function, unlike `OSSL_PARAM_UTF8_STRING`. Therefore, `default_fixup_args` is never called for `OSSL_PARAM_OCTET_STRING` and the case where `ctx->p2` is `NULL` never happens for `OSSL_PARAM_OCTET_STRING`, unlike `OSSL_PARAM_UTF8_STRING`.

Reply:
Point is that conceptually, there's not much difference between OSSL_PARAM_OCTET_STRING and OSSL_PARAM_UTF8_STRING.  They both indicate strings, the only "real" difference being their intent, and how they are normally displayed to humans.

Thanks for the detailed feedback, though.  It's possible that I missed a detail in addressing between different callbacks when I put together this module.  (like I said, it *is* a tricky module that takes a lot of historical cases into account)